### PR TITLE
Added partial filter index functionality

### PR DESCRIPTION
**Pull request by sschmittBt**

### CODE REVIEW

**Review by remi128 — CHANGES_REQUESTED** [review]
2 tests fail:

testIndexMiniMapper_partialFilter
testIndexMiniMapper_unique_filter

**Comment by sschmittBt** [comment]
I don't get these test errors myself, but I have an idea where they come from. Could you check if there are still failures with the latest commit?

File: vertx-pojongo/src/main/java/de/braintags/vertx/jomnigate/mongo/MongoUtil.java [hunk 1]
```diff
@@ -143,55 +152,95 @@ public static final void getIndexNames(final MongoDataStore ds, final String col
...
+        addIndexField(keyObject, field);
+      }
+    } catch (Exception e) {
+      future.fail(e);
+      return future;
     }
-    addIndexOptions(idxObject, indexDefinition.getIndexOptions());
-    return idxObject;
+    addIndexOptions(idxObject, indexDefinition.getIndexOptions(), mapper, dataStore, future);
+    return future;
```
Comment (remi128) [inline vertx-pojongo/src/main/java/de/braintags/vertx/jomnigate/mongo/MongoUtil.java:195]: Wouldn't it be more readable like that and less complexity?

      addIndexOptions(idxObject, indexDefinition.getIndexOptions(), mapper, dataStore, future);
    } catch (Exception e) {
      future.fail(e);
    }
    return future;

**Review by remi128 — APPROVED** [review]
ok, solved by compilersettings "store information abour method parameters"

**Pull request merged**